backend/FwLite/FwDataMiniLcmBridge/Api/FwDataMiniLcmApi.cs

```diff
@@ -304,44 +320,28 @@ public IAsyncEnumerable<Entry> SearchEntries(string query, QueryOptions? options
 
     public async Task<Entry> CreateEntry(Entry entry)
     {
-        // TODO: The API requires a value and the UI assumes it has a value. How should we handle this?
-        // if (entry.Id != default) throw new NotSupportedException("Id must be empty");
-        Guid entryId = default;
+        entry.Id = entry.Id == default ? Guid.NewGuid() : entry.Id;
         UndoableUnitOfWorkHelper.Do("Create Entry",
             "Remove entry",
             Cache.ServiceLocator.ActionHandler,
             () =>
             {
-                var stemMorphType = MorphTypeRepository.GetObject(MoMorphTypeTags.kguidMorphStem);
-                var firstSense = entry.Senses.FirstOrDefault();
-                var lexEntry = LexEntryFactory.Create(new LexEntryComponents
-                {
-                    MorphType = stemMorphType,
-                    LexemeFormAlternatives = MultiStringToTsStrings(entry.LexemeForm),
-                    GlossAlternatives = MultiStringToTsStrings(firstSense?.Gloss),
-                    GlossFeatures = [],
-                    MSA = null
-                });
+                var lexEntry = LexEntryFactory.Create(entry.Id, Cache.ServiceLocator.GetInstance<ILangProjectRepository>().Singleton.LexDbOA);
+                lexEntry.LexemeFormOA = Cache.ServiceLocator.GetInstance<IMoStemAllomorphFactory>().Create();
+                UpdateLcmMultiString(lexEntry.LexemeFormOA.Form, entry.LexemeForm);
                 UpdateLcmMultiString(lexEntry.CitationForm, entry.CitationForm);
                 UpdateLcmMultiString(lexEntry.LiteralMeaning, entry.LiteralMeaning);
                 UpdateLcmMultiString(lexEntry.Comment, entry.Note);
-                if (firstSense is not null)
-                {
-                    var lexSense = lexEntry.SensesOS.First();
-                    ApplySenseToLexSense(firstSense, lexSense);
-                }
 
-                //first sense is already created
-                foreach (var sense in entry.Senses.Skip(1))
+                foreach (var sense in entry.Senses)
                 {
                     CreateSense(lexEntry, sense);
                 }
 
-                entryId = lexEntry.Guid;
             });
-        if (entryId == default) throw new InvalidOperationException("Entry was not created");
+        if (entry.Id == default) throw new InvalidOperationException("Entry was not created");
 
-        return await GetEntry(entryId) ?? throw new InvalidOperationException("Entry was not found");
+        return await GetEntry(entry.Id) ?? throw new InvalidOperationException("Entry was not found");
```

Comment:
This error should probably be "Entry was not created" now, as the only way it would fail to be found in the entry repository is if the unit of work above failed to create it for some reason.